Bind rest server to all interfaces instead of localhost by default

SlavoKrupa opened the pull request:
* this allows running in docker images and exposing the rest endpoint
* supports openshift and other cloud providers

**Note:** test with secured endpoint have pregenerated certificates
for localhost hence the change to "localhost"

livthomas commented on http-server-microservice-provider/src/main/java/io/silverware/microservices/providers/http/HttpServerMicroserviceProvider.java:
```diff
@@ -67,7 +67,7 @@
       this.context = context;
       context.getProperties().putIfAbsent(HTTP_SERVER_PORT, 8080);
-      context.getProperties().putIfAbsent(HTTP_SERVER_ADDRESS, "localhost");
+      context.getProperties().putIfAbsent(HTTP_SERVER_ADDRESS, "0.0.0.0");
```
Why is `0.0.0.0` the default server address? I would expect either `localhost` or `127.0.0.1`.

SlavoKrupa replied:
When you specify localhost in a URL, Camel exposes the endpoint only on the local TCP/IP network interface, so it cannot be accessed from outside the machine it operates on.
If you need to expose a Jetty endpoint on a specific network interface, the numerical IP address of this interface should be used as the host. If you need to expose a Jetty endpoint on all network interfaces, the 0.0.0.0 address should be used.
Source:
http://camel.apache.org/undertow.html

livthomas replied:
Oh, thanks for the explanation.

livthomas commented on http-server-microservice-provider/src/main/java/io/silverware/microservices/providers/http/HttpServerMicroserviceProvider.java:
```diff
@@ -131,6 +131,11 @@ public void run() {
+                  readProperty(HTTP_SERVER_PORT),
+                  readProperty(HTTP_SERVER_REST_SERVLET_MAPPING_PREFIX),
+                  readProperty(HTTP_SERVER_REST_CONTEXT_PATH));
             log.info("Started Http Server.");
```
Since a new log info message has been added, this one can be removed. The new one should also be renamed from _Starting server at_ to _Started HTTP server at_ because by the time it is logged the server is already started, isn't it?

SlavoKrupa replied:
Thx. fixed

SlavoKrupa committed:
Bind rest server to all interfaces instead of localhost by default

* this allows running in docker images and exposing the rest endpoint
* supports openshift and other cloud providers
Note: test with secured endpoint have pregenerated certificates
for localhost hence the change to "localhost"
